nvme_gw_server.py

```diff
@@ -62,6 +64,19 @@ def __init__(self, nvme_config):
         self.server = None
         self.transports = set()
 
+        self.gateway_name = self.nvme_config.get("config", "gateway_name")
+        if not self.gateway_name:
+            self.gateway_name = socket.gethostname()
+
+        self.traddr = None
+        gateway_addr = self.nvme_config.get("config", "gateway_addr")
+        try:
+            ipaddr = ipaddress.ip_address(gateway_addr.strip('[]'))
```

Comment:
Do we expect arrays in `gateway_addr`? We support one only anyway, no?

Reply:
Right now in the config example we ship we have [1]:
```
gateway_addr = [::]
```
I suppose it is not unusual practice to specify an ipv6 address in `[]` and `ipaddress` has problem with parsing it, so I had to strip it.

BTW, my current patch changes `gateway_addr = [::]` to `gateway_addr = 127.0.0.1`  in the config so the test that tests "add_listener" without gateway would work by default.

[1] https://github.com/ceph/ceph-nvmeof/blob/devel/nvme_gw.config#L13

Reply:
Ok it seems like the `[]` is used to distinguish between colons used after an address to specify the port in an URI [1], e.g. `[2001:db8:85a3:8d3:1319:8a2e:370:7348]:443`. I don't think we need to support `[]` syntax here.

[1]: [https://datatracker.ietf.org/doc/html/rfc6874](https://datatracker.ietf.org/doc/html/rfc6874)